lib/huntaway.rb

```diff
@@ -9,22 +9,31 @@ class Huntaway
   FIRST_LINE_DEV_SUPPORT_GROUP_ID = ENV.fetch("FIRST_LINE_DEV_SUPPORT_GROUP_ID").to_i
   OPSGENIE_SCHEDULE_ID = ENV.fetch("OPSGENIE_SCHEDULE_ID")
 
+  @@existing_group_memberships = []
+
   def run!
-    unassign_users_from_group
-    assign_current_support_users_to_group
+    unassign_extra_users_from_group
+    assign_missing_support_users_to_group
   end
 
   private
 
-  def assign_current_support_users_to_group
+  def assign_missing_support_users_to_group
     current_support_user_ids.each do |id|
-      client.group_memberships.create!(user_id: id, group_id: FIRST_LINE_DEV_SUPPORT_GROUP_ID)
+      unless @@existing_group_memberships.include?(id)
```

Comment:
What's the rationale behind making this a class variable?

Reply:
A class variable means we can populate it as part of the process of deleting (which already includes the logic for if a membership is part of the right group or not). I suppose we could easily enough pull it out into a method and isolate that bit of logic.

Reply:
Yeah, but it's all taking place within one instance of the same class, so only needs to be an instance variable right?